components/ai_chat/resources/page/components/alerts/error_rate_limit.tsx

```diff
@@ -20,18 +20,24 @@ function ErrorRateLimit() {
     // Freemium model with non-premium user has stricter rate limits. Secondary
     // action is to switch to completely free model.
     if (context.currentModel?.access === mojom.ModelAccess.BASIC_AND_PREMIUM) {
+      const handleSwitchToBasicModel = () => {
+        context.switchToBasicModel()
+        getPageHandlerInstance().pageHandler.retryAPIRequest()
```

Comment:
Is this guaranteed to happen in the exact same order in c++?

Reply:
switching model writes to `model_key_`, initializes engine, which aborts current pending requests and at the end notifies that model changed and api progress is none. retry api requests immediately resets the current error to none, finds the last human query and submits the conversation.

the initialization and notification of the model change seem pretty sync and deterministic, and we retry the request afterward. so, i don't see why it doesn't guarantee order here.

Reply:
According to https://github.com/chromium/chromium/blob/main/docs/mojo_ipc_conversion.md order is guranteed